lib/Sema/SemaExpr.cpp

```diff
@@ -13919,14 +13940,26 @@ void Sema::MarkFunctionReferenced(SourceLocation Loc, FunctionDecl *Func,
     } else if (MethodFunName == "__cxxamp_trampoline"||
          MethodFunName == "__cxxamp_trampoline_name") {
       DefineAMPTrampoline(Loc, MethodDecl);
-    } else if (MethodDecl->isOverloadedOperator() &&
-               MethodDecl->getOverloadedOperator() == OO_Equal) {
-      MethodDecl = cast<CXXMethodDecl>(MethodDecl->getFirstDecl());
-      if (MethodDecl->isDefaulted() && !MethodDecl->isDeleted()) {
-        if (MethodDecl->isCopyAssignmentOperator())
-          DefineImplicitCopyAssignment(Loc, MethodDecl);
-        else if (MethodDecl->isMoveAssignmentOperator())
-          DefineImplicitMoveAssignment(Loc, MethodDecl);
+    } else if (MethodDecl->isOverloadedOperator()) {
+      if (is_hip_functor(MethodDecl)) { // TODO: temporary.
+        const auto b = cast<CompoundStmt>(MethodDecl->getBody());
+        auto p = std::find_if(
+            b->body_begin(),
+            b->body_end(),
+            [](Stmt* x) { return isa<CallExpr>(x); });
+        if (p != b->body_end()) {
```

Comment:
I guess the assumption here is that the first call in the manufactured functor will always be a call to an HIP global function?  If that's the case, could we add an assert here to ensure this condition is met?